packages/client/lib/rpc/modules/eth.ts

```diff
@@ -273,15 +274,28 @@ export class Eth {
   }
 
   /**
-   * Returns information about a block by hash.
+   * Returns information about a block by block number.
    * @param params An array of two parameters:
-   *   1. a block hash
+   *   1. integer of a block number, or the string "latest", "earliest" or "pending"
    *   2. boolean - if true returns the full transaction objects, if false only the hashes of the transactions.
    */
-  async getBlockByHash(params: [string, boolean]) {
-    const [blockHash, includeTransactions] = params
+  async getBlockByNumber(params: [string, boolean]) {
+    const [blockOpt, includeTransactions] = params
+    let block: Block
+    if (blockOpt === 'latest') {
+      block = await this._chain.getLatestBlock()
+    } else if (blockOpt === 'pending') {
+      return {
+        code: INVALID_PARAMS,
+        message: `"pending" is not yet supported`,
+      }
+    } else if (blockOpt === 'earliest') {
+      block = await this._chain.getBlock(new BN(0))
+    } else {
+      const blockNumberBN = new BN(stripHexPrefix(blockOpt), 16)
+      block = await this._chain.getBlock(blockNumberBN)
+    }
```

Comment:
```suggestion
    } else if (blockOpt.substr(0, 2) === '0x') {
      const blockNumberBN = new BN(stripHexPrefix(blockOpt), 16)
      block = await this._chain.getBlock(blockNumberBN)
    } else {
      return {
        message: `invalid argument ${index}: argument must be a hex string`,
        code: INVALID_PARAMS,
      }
    }
```

Reply:
Thanks!

Shouldn't that case be always covered by the [`blockOption` validation here](https://github.com/gabrocheleau/ethereumjs-monorepo/blob/feat/extend-getBlockByNumber/packages/client/lib/rpc/validation.ts#L126-L154)